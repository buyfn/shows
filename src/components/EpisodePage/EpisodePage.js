import React from 'react';

class EpisodePage extends React.Component {
  componentDidMount() {
    const episodeID = this.props.match.params.episodeID;
    this.props.fetchEpisode(episodeID);
  }

  render() {
    const { image, name, summary, fetchingState, errorMsg } = this.props;

    if (fetchingState === 'fetching') {
      return 'Fetching data...';
    }

    if (fetchingState === 'failed') {
      return errorMsg;
    }

    return (
      <>
        {image && (
          <img alt={`"${name}" episode cover`} src={image.original} />
        )}

        <h1>{name}</h1>

        <div dangerouslySetInnerHTML={{ __html: summary }} />
      </>
    );
  }
}

export default EpisodePage;
