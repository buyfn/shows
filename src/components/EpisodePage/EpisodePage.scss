.episode-page {
    max-width: 1400px;
}

.episode-cover {
    width: 100%;
}

.episode-image {
    width: 100%;
    object-fit: cover;
}

.episode-info {
    padding: 0 2em 4em;
}