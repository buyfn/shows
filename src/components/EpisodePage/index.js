import { connect } from 'react-redux';

import { fetchEpisodeRequest } from '../../actions';

import EpisodePage from './EpisodePage';

const mapStateToProps = (state) => ({
  ...state.episode,
});

const mapDispatchToProps = dispatch => ({
  fetchEpisode: (id) => dispatch(fetchEpisodeRequest(id)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(EpisodePage);
