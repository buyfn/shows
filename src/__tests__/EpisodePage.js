import React from 'react';
import renderer from 'react-test-renderer';

import EpisodePage from '../components/EpisodePage/EpisodePage';

describe('EpisodePage', () => {
  const props = {
    match: {
      params: { episodeID: '1' }
    },
    summary: 'test episode summary',
    name: 'test episode name',
    image: {
      medium: 'img.jpg',
      original: 'img.jpg'
    },
  };

  let tree;
  const fetchEpisodeMock = jest.fn();

  beforeAll(() => {
    tree = renderer
      .create(<EpisodePage {...props} fetchEpisode={fetchEpisodeMock} />)
      .toJSON();
  });

  it('calls "fetchShow" with correct argument', () => {
    expect(fetchEpisodeMock).toHaveBeenCalledWith('1');
    expect(fetchEpisodeMock.mock.calls.length).toEqual(1);
  });

  it('renders correctlly', () => {
    expect(tree).toMatchSnapshot();
  });
});
