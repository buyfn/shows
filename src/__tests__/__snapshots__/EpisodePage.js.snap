// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodePage renders correctlly 1`] = `
Array [
  <img
    alt="\\"test episode name\\" episode cover"
    src="img.jpg"
  />,
  <h1>
    test episode name
  </h1>,
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "test episode summary",
      }
    }
  />,
]
`;
